### ISSUE TITLE
Foxy migration

### ISSUE THREAD

**Issue by pablogs9** [micro-ROS/micro-ros-build#131]
This issue reports all repos involved in foxy migration and details the future requirements for each of them.

micro-ROS Foxy migration is compound of multiple repos with `feature/foxy_migration` branches that must be merged into a final `foxy` branch on each of them:

| Repo                                      | Pre-release branch                                                                                                 | Comments                                      | Release PR                                                           |
| ----------------------------------------- | ------------------------------------------------------------------------------------------------------------------ | --------------------------------------------- | -------------------------------------------------------------------- |
| micro-ROS/micro-ros-build                 | [feature/foxy_migration](https://github.com/micro-ROS/micro-ros-build/tree/feature/foxy_migration)                 |                                               | https://github.com/micro-ROS/micro-ros-build/pull/148                                                              |
| micro-ROS/rcl                             | [feature/foxy_migration](https://github.com/micro-ROS/rcl/tree/feature/foxy_migration)                             | Should be merged into ros2/rcl                | https://github.com/micro-ROS/rcl/pull/9                                                             |
| micro-ROS/rcutils                         | [feature/foxy_migration](https://github.com/micro-ROS/rcutils/tree/feature/foxy_migration)                         | Should be merged into ros2/rcutils            | https://github.com/micro-ROS/rcutils/pull/8/                                                              |
| micro-ROS/rosidl_typesupport              | [feature/foxy_migration](https://github.com/micro-ROS/rosidl_typesupport/tree/feature/foxy_migration)              | Should be merged into ros2/rosidl_typesupport |https://github.com/micro-ROS/rosidl_typesupport/pull/1                                                              |
| micro-ROS-demos                           | [feature/foxy_migration](https://github.com/micro-ROS/micro-ROS-demos/tree/feature/foxy_migration)                 |                                               | https://github.com/micro-ROS/micro-ROS-demos/pull/25                 |
| micro-ROS/rosidl_typesupport_microxrcedds | [feature/foxy_migration](https://github.com/micro-ROS/rosidl_typesupport_microxrcedds/tree/feature/foxy_migration) |                                               | https://github.com/micro-ROS/rosidl_typesupport_microxrcedds/pull/18 |
| micro-ROS/rmw-microxrcedds                | [feature/foxy_migration](https://github.com/micro-ROS/rmw-microxrcedds/tree/feature/foxy_migration)                |                                               | https://github.com/micro-ROS/rmw-microxrcedds/pull/70                |
| micro-ROS/apps                            | [feature/foxy_migration](https://github.com/micro-ROS/apps/tree/feature/foxy_migration)                            |                                               | https://github.com/micro-ROS/apps/pull/25                                                              |
| micro-ROS/freertos_apps                   | [feature/foxy_migration](https://github.com/micro-ROS/freertos_apps/tree/feature/foxy_migration)                   |                                               | https://github.com/micro-ROS/freertos_apps/pull/13/files                                                              |
| micro-ROS/raspbian_apps                   | [feature/foxy_migration](https://github.com/micro-ROS/raspbian_apps/tree/feature/foxy_migration)                                                                  |                                               | https://github.com/micro-ROS/raspbian_apps/pull/6                                                              |
| micro-ROS/micro-ROS-Agent                 | [feature/foxy_migration](https://github.com/micro-ROS/micro-ROS-Agent/tree/feature/foxy_migration)                 | Check required changes                        | https://github.com/micro-ROS/micro-ROS-Agent/pull/30                                                              |
| micro-ROS/zephyr_apps                     | [feature/foxy_migration](https://github.com/micro-ROS/zephyr_apps/tree/feature/foxy_migration)                     |                                               |  https://github.com/micro-ROS/zephyr_apps/pull/11            |
| micro-ROS/ros_tracing                     | [feature/foxy_migration](https://gitlab.com/micro-ROS/ros_tracing/ros2_tracing/-/tree/feature/foxy_migration)      |                                               | https://gitlab.com/micro-ROS/ros_tracing/ros2_tracing/-/merge_requests/192                                                              |
| micro-ROS/microros-github.io              | Pending                                                                                                            |                                               | Pending                                                              |
| micro-ROS/docker                          | [feature/foxy_migration](https://github.com/micro-ROS/docker/pull/46)                                              |                                               | https://github.com/micro-ROS/docker/pull/46                          |

_Before release every row must have a PR merged from `feature/foxy_migration` to `foxy`_

Related PR, should be merged before release procedure:

- [x] https://github.com/micro-ROS/rclc/pull/33
- [x] https://github.com/micro-ROS/rclc/pull/32
- [x] https://github.com/micro-ROS/zephyr_apps/pull/10
- [x] https://github.com/micro-ROS/zephyr_apps/pull/9
- [x] https://github.com/micro-ROS/zephyr_apps/pull/8
- [x] https://github.com/micro-ROS/zephyr_apps/pull/10
- [x] https://github.com/micro-ROS/micro-ros-build/pull/130
- [x] https://github.com/micro-ROS/apps/pull/23
- [x] https://github.com/micro-ROS/rmw-microxrcedds/pull/64
- [x] https://github.com/micro-ROS/rmw-microxrcedds/pull/65
- [x] https://github.com/micro-ROS/rmw-microxrcedds/pull/66
- [x] https://github.com/micro-ROS/rmw-microxrcedds/pull/67
- [x] https://github.com/micro-ROS/freertos_apps/pull/6
- [x] https://github.com/micro-ROS/rmw-microxrcedds/pull/68
- [x] https://github.com/micro-ROS/micro-ros-build/pull/132
- [x] https://github.com/micro-ROS/rcl/pull/7
- [x] https://github.com/micro-ROS/micro-ros-build/pull/133
- [x] https://github.com/micro-ROS/micro-ROS-Agent/pull/30
- [x] https://github.com/micro-ROS/micro-ros-build/pull/135
- [x] https://github.com/micro-ROS/rcutils/pull/7
- [x] https://github.com/micro-ROS/rmw-microxrcedds/pull/69
- [x] https://github.com/micro-ROS/rosidl_typesupport_microxrcedds/pull/17
- [x] https://github.com/micro-ROS/rosidl_typesupport_microxrcedds/pull/21

NOTES:
- https://github.com/eProsima/crazyflie-clients-python should be a standalone CF Radio to micro-ROS Agent bridge with installation instructions: WIP

- All `feature/foxy_migration` branches are referenced from this micro-ROS Build System repo. The last pre-release commits must change all these branches from `feature/foxy_migration` to `foxy`

POST RELEASE:
- Parameters and graph support will be released just after micro-ROS Foxy Release 
- [x] `rcutils` repository HEAD commit has changed in origin ros2 repository since the moment it was forked. A further revision of the last  introduced changes must be done before rebasing this repository to ros2:foxy branch.
- [x] repository trace_tools need a review in the CI in order to merge to foxy
- Check https://github.com/eProsima/Micro-XRCE-DDS-Client/tree/feature/wait_until_data
- [x] Change main branch to foxy for all repos

**Comment by MillerLuca** [comment]
Hello,

You are doing a great job with the development of Micro-ROS, 
We are wondering when we should expect a Foxy release of Micro ROS ? 

Thank you for your hard work,
Luca Miller

**Comment by pablogs9** [comment]
Hi thank!
We are currently working on it. I hope it is ready for the next week.

**Comment by MillerLuca** [comment]
Great !!
Looking forward to play around with it,
Thank you for your hard work ;-)

**Comment by russkel** [comment]
This looks great. Is this complete now @pablogs9 ?

**Comment by pablogs9** [comment]
Yes, Foxy release is out!
